API/twitter/TweetExtractor.py

```diff
@@ -45,118 +53,160 @@ def scrape_tweet(query,count=50,earliest=None,latest=None):
+            'Likes' : i_likes,
+            'Location' : s_location,
+            'Link' : s_url
         })
 
-        #Checking for media in the tweet
-        if hasattr(tweets[i], "extended_entities"):
-            count = 1
-            #for all media in tweet
-            for photo in tweets[i].extended_entities["media"]:
-                #saving media link and file name
-                if (photo['type'] == "photo"):
-                    picture_name = f"{media_directory}pictureID#{str(i).zfill(5)}-{str(count)}.jpg"
-                    link = photo['media_url_https']
-                elif (photo['type'] == "video"):
-                    picture_name = f"{media_directory}videoID#{str(i).zfill(5)}-{str(count)}.mp4"
-                    link = photo['video_info']['variants'][0]['url']
-                elif (photo['type'] == "animated_gif"):
-                    picture_name = f"{media_directory}gifID#{str(i).zfill(5)}-{str(count)}.mp4"
-                    link = photo['video_info']['variants'][0]['url']
-
-                #download media
-                r = requests.get(link)
-                with open(picture_name, 'wb') as f:
-                    f.write(r.content)
-
-                count += 1
+        #Checking for media in the extended_entities attribute of tweet object
+        #More info about the extended_entities attribute can be found here:
+        #https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/extended-entities
+        if hasattr(l_tweets[i], "extended_entities"):
+            i_media_count = 1       #integer, used to index media names should the tweet object have more than one
+            #for each piece of media in extended entities
+            for o_media in l_tweets[i].extended_entities["media"]:
+                #build name media will be saved as
+                s_media_name = f"{s_media_directory}pictureID#{str(i_tweet_id).zfill(5)}-{str(i_media_count)}"
+
+                #find media type, append the associated file type to end of s_media_name, and save link to media in s_media_link
+                if (o_media['type'] == "photo"):
+                    s_media_name = s_media_name + ".jpg"
+                    s_media_link = o_media['media_url_https']
+                elif (o_media['type'] == "video"):
+                    s_media_name = s_media_name + ".mp4"
+                    s_media_link = o_media['video_info']['variants'][0]['url']
+                elif (o_media['type'] == "animated_gif"):
+                    #gif's are saved as .mp4's because that's how twitter stores them, as shown in the extended_entities object link above
+                    s_media_name = s_media_name + ".mp4"
+                    s_media_link = o_media['video_info']['variants'][0]['url']
+
+                #request s_media_link and download it's contents to s_media_name
+                o_request = requests.get(s_media_link)
+                with open(s_media_name, 'wb') as o_file:
```

Comment:
I feel like o_file doesn't follow our convention well. f_media would make more sense since it's a file that contains media.